src/pk/asn1/der/sequence/der_length_sequence.c

```diff
@@ -50,6 +50,9 @@ int der_length_sequence_ex(ltc_asn1_list *list, unsigned long inlen,
           break;
        }
 
+       /* some items may be optional during import */
+       if (!list[i].used && list[i].optional) continue;
```

Comment:
most likely a dumb comment, but ... we're taking the `optional` flag here into account and this function is called as well from `der_encode_sequence_ex()` where it is used to determine if we would BOF but `der_encode_sequence_ex()` itself ignores the `optional` flag...